pyterrier/bootstrap.py

```diff
@@ -1,10 +1,120 @@
 from . import mavenresolver
+from typing import Union, List
 
 stdout_ref = None
 stderr_ref = None
 TERRIER_PKG = "org.terrier"
 SAVED_FNS=[]
 
+class IndexFactory:
+    """
+    This class "shades" the native Terrier `IndexFactory <http://terrier.org/docs/current/javadoc/org/terrier/structures/IndexFactory.html>`_ class - it offers essential the same API,
+    except that the ``of()`` method contains a memory kwarg, that can be used to load additional index data structures into memory. 
+    """
+
+    @staticmethod
+    def _load_into_memory(index, structures=['lexicon', 'direct', 'inverted', 'meta'], load=False):
```

Comment:
What about document index? It is accessed for getting doc lens, so quite often during ranking.

Reply:
document lengths are always in memory. the choice of memory or not relates to the other information, specifically the bits used to access the direct index